### PR TITLE
Add MP4 to Transcode API

### PR DESCRIPTION
**Pull request by leszko**
It also updates the Catalyst API VOD according to the design here: https://www.notion.so/livepeer/VOD-MP4-separate-from-HLS-34efd71b091a491ea4e785dda3bf9873

### CODE REVIEW

File: task/paths.go
Comment (thomshutt) [inline task/paths.go:20]: Is this method doing anything now?

Comment (leszko) [reply task/paths.go]: hehe, right, removed.

File: task/upload.go
Comment (thomshutt) [inline task/upload.go:530]: Could we have these as constants?

File: task/upload.go
Comment (thomshutt) [inline task/upload.go:552]: You didn't introduce the method, so looks like it was already untested but given we have a fairly discrete set of inputs and outputs it should be pretty unit testable - would you mind adding a test in here?

**Review by thomshutt — APPROVED** [review]
Other than my comments this LGTM - I don't think either of them is a blocker, so your call on addressing them

File: clients/catalyst.go
Comment (victorges) [inline clients/catalyst.go:50]: Wow is this already supported?

File: task/upload.go
Comment (victorges) [inline task/upload.go:552]: **niiiit:** can drop a few characters here lol
```suggestion
func outputLocations(outURL, hls, hlsRelPath, mp4, mp4RelPath string) ([]OutputName, []clients.OutputLocation, error) {
```

File: task/upload.go
Comment (victorges) [inline task/upload.go:567]: Just double-checking: Does catalyst support multiple `output_locations` provided on the request now? Previously it only read the first one and checked the outputs

File: task/upload.go
Comment (victorges) [inline task/upload.go:558]: You have to add another value to this slice here. Let me share a bit more context of why this `outputLocations` function exists (my bad for not having documented it originally, this was all done in a rush): 
 - The original Catalyst API provided no way to associated outputs from the requested output_locations
 - It actually ended up supporting only 1 location and 1 output in the first version, but it received and returned an array
 - There would also be other output types that would be incompatible with sending everything on a single output_location: e.g. saving to IPFS, which would have a completely different URL with pinata creds instead of S3

So we created this function here, which would create outputs in a deterministic order and then we did a logic on `processCatalystCallback` that assumes Catalyst also returns the `outputs` on the exact order as they were requested [1]. That's the whole reason for having those `names` here which are also returned and then processed in order [on the big switch in the process result function](https://github.com/livepeer/task-runner/blob/b112d7458c808c7d01f54afdab2f06ccae706368/task/upload.go#L332).

[1] (but we don't actually use the same request data from the task params, we regenerate the outputs which is error prone in case of deploys)

I'm not sure if either of those constraints are still true. Maybe we can already associate specific `output_locations` with their corresponding `outputs` now. Or maybe the output data became different enough that we can actually tell which is which without trying to associate.

Either way, I never liked this logic, so feel free to refactor it completely. For now though, since you are adding a new output location, you might need to add a new output name and make sure that `processCatalystCallback` is able to process it consistently. In the long run it'd be good to have a safer way to associate the 2 things though, like an output_location id, or maybe the outputs just embed the entire output_location requests, idk!

File: clients/catalyst.go
Comment (leszko) [reply clients/catalyst.go:50]: Yeah, in the other PR: https://github.com/livepeer/catalyst-api/pull/546

File: task/upload.go
Comment (leszko) [reply task/upload.go:558]: Ok, I get it now. It's very confusing, for now, I'll add another output here, but we should think about tidying it up a little bit.

File: task/upload.go
Comment (leszko) [reply task/upload.go:530]: Sure, moved to constants

File: task/upload.go
Comment (leszko) [reply task/upload.go:574]: Yes, it will support it after this PR is merged: https://github.com/livepeer/catalyst-api/pull/546

File: task/upload.go
Comment (leszko) [reply task/upload.go]: Added unit tests.

**Review by mjh1 — APPROVED** [review]

File: task/upload.go
Comment (victorges) [reply task/upload.go]: Yeah agreed. We need some reliable way to map outputs to the requests somehow

**Pull request merged**
